src/spells.cpp

```diff
@@ -236,7 +236,8 @@ uint32_t Spells::getInstantSpellCount(const Player* player) const
 
 InstantSpell* Spells::getInstantSpellById(uint32_t spellId)
 {
-	auto& it = std::advance(instants.begin(), spellId);
+	auto& it = instants.begin();
+	std::advance(it, spellId);
```

Comment:
As I said, std::advance or std::next past the end of the container will give you undefined behavior:
```cpp
int main() {
	std::map<int32_t, int32_t> test = {
		{1, 2},
		{3, 4},
		{5, 6}
	};
	auto it = std::next(test.begin(), 12);
	if (it != test.end()) {
		std::cout << it->second << std::endl;
	}
	return 0;
}
```
This is printing 6 for me compiling with gcc.